app/mender_test.go

```diff
@@ -1243,15 +1254,24 @@ func TestMutualTLSClientConnection(t *testing.T) {
 [{ "time": "12:12:12", "level": "error", "msg": "log foo" },
 { "time": "12:12:13", "level": "debug", "msg": "log bar" }]
 }`)
-
-			err = mender.UploadLog(
-				&datastore.UpdateInfo{
-					ID: "foobar",
-				},
-				logs,
-			)
-			test.assertFunc(t, err, srv.Log.Logs)
-
+			var t2 *testObject
+			assert.Eventually(t, func() bool {
+				t2 = &testObject{}
+				err = mender.UploadLog(
+					&datastore.UpdateInfo{
+						ID: "foobar",
+					},
+					logs,
+				)
+				t2.failed = false
+				test.assertFunc(t2, err, srv.Log.Logs)
+				return !t2.failed
+			}, 10*time.Second, 500*time.Millisecond)
```

Comment:
maybe we could define those timeouts somewhere above as vars or in the test case.

Reply:
Done!

Reply:
lovely!